packages/amplify-cli-core/src/jsonUtilities.ts

```diff
@@ -0,0 +1,126 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as hjson from 'hjson';
+
+export class JSONUtilities {
+  public static readJson = async <T>(
+    fileName: string,
+    options: {
+      throwIfNotExist?: boolean;
+      preserveComments?: boolean;
+    } = {
+      throwIfNotExist: true,
+      preserveComments: false,
+    },
+  ): Promise<T> => {
+    if (!fileName) {
+      throw new Error(`'fileName' argument missing`);
+    }
+
+    if (!fs.existsSync(fileName)) {
+      if (options.throwIfNotExist === true) {
+        throw new Error(`File at path: '${fileName}' does not exist`);
+      } else {
+        return (undefined as unknown) as T;
```

Comment:
I'd change the return type to Promise<T | undefined>. Makes the function more explicit that it could return undefined.